src/main.cpp

```diff
@@ -2456,18 +2507,32 @@ void static UpdateTip(CBlockIndex *pindexNew) {
     {
         int nUpgraded = 0;
         const CBlockIndex* pindex = chainActive.Tip();
+        for (int bit = 0; bit < 29; bit++) {
+            WarningBitsConditionChecker checker(bit);
+            ThresholdState state = checker.GetStateFor(pindex, chainParams.GetConsensus(), warningcache[bit]);
+            if (state == THRESHOLD_ACTIVE || state == THRESHOLD_LOCKED_IN) {
+                if (state == THRESHOLD_ACTIVE) {
+                    strMiscWarning = strprintf(_("Warning: unknown new rules activated (versionbit %i)"), bit);
+                    CAlert::Notify(strMiscWarning, true);
+                    fWarned = true;
+                } else {
+                    LogPrintf("%s: unknown new rules are about to activate (versionbit %i)\n", bit);
+                }
+            }
+        }
+        int32_t nExpectedVersion = ComputeBlockVersion(pindex->pprev, chainParams.GetConsensus());
```

Comment:
(Sorry if this is a repost, I think github ate my first attempt to comment)

I think we should move this line into the for loop below; otherwise we'd generate a warning after a known soft-fork activates.